src/main/java/javaslang/control/ValidationBuilder.java

```diff
@@ -0,0 +1,75 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+import javaslang.Function2;
+import javaslang.Function3;
+import javaslang.Function4;
+import javaslang.collection.List;
+
+public class ValidationBuilder<E,T1,T2> {
```

Comment:
No, it needs to be public/accessible for callers of the API. We place it in the Validation interface.

This is similar to all the nested Match interfaces. We use this pattern in many places (Try.Success/Failure, Option.Some/None, ...). It can be seen as some kind of internal DSL. The Builders are needed to technically ensure that we get our types right but in first place they are not needed by our _business domain_ (which is also technical here but that's only a matter of abstraction).